[LTS] Use public client auth for odsp e2e flows (#21091)

Abe27342 opened the pull request:
## Description

Currently, we use the ropc flow with a confidential client to authenticate in our e2e/stress tests against odsp. This is less appropriate than using a public client flow, since our application registration really only needs to have delegated permissions.
This PR adjusts things to use a public flow--I've updated the relevant app registrations to allow both forms of auth already.

Using a public flow also means our infrastructure setup here aligns exactly with [this relatively recent MSFT guidance](https://learn.microsoft.com/en-us/entra/identity-platform/test-automate-integration-testing?tabs=dotnet).

Cherry-pick of #21091. This change needs to be in every branch we plan on maintaining real service tests against odsp for, as I plan on removing the application configuration enabling confidential client auth.

Commit:
Use public auth for odsp tests

Commit:
Spot fix odspAuth.ts

Commit:
Remove stray references to login__microsoft__secret

Abe27342 commented:
needed to make a bunch of manual updates here b/c of differences in formatting etc.

Abe27342 commented on packages/utils/odsp-doclib-utils/src/odspAuth.ts:
```diff
@@ -13,6 +13,13 @@ export interface IOdspTokens {
+/**
+ * Configuration for a public client.
+ */
+export interface IPublicClientConfig {
```
typetests don't support removing interfaces on LTS.

markfields replied:
Deprecate it? Not that it matters

Abe27342 replied:
yeahhh this is all internal in modern code anyway, so probably we did that at some point.

Commit:
fix lint in tool-utils

markfields reviewed (approved):
Diff looks fine.  You could also do global search for interesting keywords like `login__microsoft__secret` to make sure there aren't other hits to deal with

Abe27342 commented:
> Diff looks fine. You could also do global search for interesting keywords like `login__microsoft__secret` to make sure there aren't other hits to deal with

yup. I did that. there was in fact a file hanging around which I removed a ref from. (though I think it was one we deleted since it was long unused so probably nbd if I didn't find the reference)
